vulcanai/models/metrics.py

```diff
@@ -892,12 +893,41 @@ def _boot_cv(network, data_loader, k, epochs, retain_graph, valid_interv,
             logger.info(
                 "\n\n***KeyboardInterrupt: Cross validate stopped \
                 prematurely.***\n\n")
-	
+
+    @staticmethod
+    def stratified_split(dataset, k, strata_column):
+
+        if strata_column == "class_label":
+            sr = pd.Series(dataset.tensors[1].numpy())
+        else:
+            #just let it fail if it doesn't work
+            sr = pd.Series(dataset[0][strata_column].numpy())
+
+        ls = [[] for i in range(k)]
+
+        grps = sr.groupby(by=sr)
+
+        for key, grp in grps:
+            inds = np.array(grp.index)
+            np.random.shuffle(inds)
```

Comment:
just want to double check this is done in-place?

Reply:
it defs is.... because I also assume it's not and then mess everything up. not this time!

Reply:
amazing! glad to hear it :)